contracts/protocols/polygon/qi_dao/main.sol

```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.8.0;
+
+import "./helpers.sol";
+import "./interfaces.sol";
+
+contract Resolver is QiDaoHelpers {
+    function getVaultCollateral(address vault, uint256 vaultID) public view returns (uint256) {
+        erc20StablecoinInterface vaultContract = erc20StablecoinInterface(vault);
+        return vaultContract.vaultCollateral(vaultID);
+    }
+
+    function getVaultDebt(address vault, uint256 vaultID) public view returns (uint256) {
+        erc20StablecoinInterface vaultContract = erc20StablecoinInterface(vault);
+        return vaultContract.vaultDebt(vaultID);
+    }
+
+    function getVaultByOwnerIndex(
+        address vault,
+        address owner,
+        uint256 idx
+    ) public view returns (uint256) {
+        erc20StablecoinInterface vaultContract = erc20StablecoinInterface(vault);
+        return vaultContract.tokenOfOwnerByIndex(owner, idx);
+    }
+}
+
```

Comment:
Why don't we create a single function which returns all details about user positions.
```suggestion
    function getPosition(
        address vault,
        address owner,
        uint256 vaultId,
        uint256 idx
    ) public view returns (uint256 debt, uint256 collateral, uint256 tokenOwner) {
        erc20StablecoinInterface vaultContract = erc20StablecoinInterface(vault);
        debt = vaultContract.vaultDebt(vaultID);
        tokenOwner = vaultContract.tokenOfOwnerByIndex(owner, idx);
        collateral = vaultContract.vaultCollateral(vaultID);
    }
```